src/components/AddSlots/b.js

```diff
@@ -0,0 +1,215 @@
+import React, { useState, useEffect, useRef } from 'react';
+import axios from 'axios';
+import {
+  Button,
+  ModalBody,
+  Modal,
+  ModalHeader,
+  ModalFooter,
+  ListGroup,
+  ListGroupItem,
+  Input
+} from 'reactstrap';
+
+import Alerts from '../Alerts';
+import { API_URL } from '../../common/consts';
+
+
+export default function AddSlots({
+  toggleAddSlots,
+  user,
+  openModal,
+  ...others
+}) {
+  let maxPeopleAllowedRef = useRef(null);
+  const [allSlots, setAllSlots] = useState([]);
+  const [error, setError] = useState({});
+  const [selectedTime, setSelectedTime] = useState({
+    start_minutes: 0,
+    start_hours: 0,
+    end_hours: 0,
+    end_minutes: 0,
+    maximun_people_allowed: 5
+  });
+
+  useEffect(() => {
+    getSlots();
+  }, []);
+
+  const showError = (type, message) => setError({ type, message });
+  const closeError = () => setError({ type: '', message: '' });
+
+
+  const getSlots = () => {
+    const storeId = user && user.storeId;
+    const filter = { where: { storeId: storeId } };
+    axios
+      .get(
+        `${API_URL}/stores_slots?filter=${JSON.stringify(
+          filter
+        )}`
+      )
+      .then(res => {
+        setAllSlots(res.data);
+      })
+      .catch(err => {
+        showError('danger', 'Error in fetching all slots');
+      });
+  };
+
+  const deleteSlots = slotId => {
+    closeError();
+    axios
+      .delete(
+        `${API_URL}/stores_slots/${slotId}`
+      )
+      .then(res => {
+        getSlots();
+      })
+      .catch(err => {
+        showError('danger', 'Error in deleting slot');
+      });
+  };
+
+  const addSlots = timeString => {
+    closeError();
+    const storeId = user && user.storeId;
+
+    const timeArray = timeString.split(' - ');
```

Comment:
I did write a function to format time strings in the helpers folder for cleanup if you want to revisit.